src/Workspaces/Core/MSBuild/MSBuild/SerializationExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.Serialization;
+using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Options;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.MSBuild
+{
+    internal static class SerializationExtensions
+    {
+        public static ProjectStateChecksums GetCheckSum(this ProjectInfo projectInfo, ISerializerService serializer)
```

Comment:
Can't say I'm a fan of the duplication of the checksumming logic. See other comment that I think this is unnecessary -- the methods this is all guarding should already handle the case just fine where you're not changing anything.